channels/channels.go

```diff
@@ -0,0 +1,87 @@
+package channels
+
+import (
+	"net/http"
+	"strings"
+
+	"github.com/jaimeteb/chatto/channels/messages"
+	"github.com/jaimeteb/chatto/channels/rest"
+	"github.com/jaimeteb/chatto/channels/slack"
+	"github.com/jaimeteb/chatto/channels/telegram"
+	"github.com/jaimeteb/chatto/channels/twilio"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/viper"
+)
+
+// Config struct combines all available client configurations
+type Config struct {
+	Telegram telegram.Config `mapstructure:"telegram"`
+	Twilio   twilio.Config   `mapstructure:"twilio"`
+	Slack    slack.Config    `mapstructure:"slack"`
+}
+
+// Channels combines all available channel clients
+type Channels struct {
```

Comment:
Declare all our `Channels` here and ensure they conform to the `Channel` interface. This allowed us to reduce duplicate code in the `bot` package.